include/wlr/types/wlr_drm_lease_v1.h

```diff
@@ -0,0 +1,129 @@
+/*
+ * This an unstable interface of wlroots. No guarantees are made regarding the
+ * future consistency of this API.
+ */
+#ifndef WLR_USE_UNSTABLE
+#error "Add -DWLR_USE_UNSTABLE to enable unstable wlroots features"
+#endif
+
+#ifndef WLR_TYPES_WLR_DRM_LEASE_V1_H
+#define WLR_TYPES_WLR_DRM_LEASE_V1_H
+
+#include <wayland-server.h>
+#include <wlr/backend.h>
+
+struct wlr_drm_backend;
+
+struct wlr_drm_lease_device_v1 {
+	struct wl_list resources; // wl_resource_get_link
+	struct wl_global *global;
+	struct wlr_drm_backend *backend;
+
+	struct wl_list connectors; // wlr_drm_lease_connector_v1::link
+	struct wl_list leases; // wl_resource_get_link
+	struct wl_list requests; // wl_resource_get_link
+
+	struct wl_listener display_destroy;
```

Comment:
Instead of listening for the `wl_display` destroy signal, we should probably listen for the backend destroy signal here.

Reply:
fixed in https://github.com/swaywm/wlroots/pull/2929/commits/232c7ba09b7fd09d3f145b14fb0c7c725f0d6843